### PR TITLE
Update quartz version and implement new methods in MongoDBJobStore

### PR DESCRIPTION
**Pull request by Chevron94**
Implementation for https://github.com/michaelklishin/quartz-mongodb/issues/157

### CODE REVIEW

**Comment by pwojnowski** [comment]
Sorry for such a delay, but I couldn't find the time to manually verify that it actually works. Could you add some tests for this code? I read that and it looks ok, but has no tests. :cry:

**Comment by Chevron94** [comment]
I added test for resetTriggerFromErrorState method

File: build.gradle [hunk 1]
```diff
@@ -50,7 +50,7 @@ idea {
     }
 }
 dependencies {
-    compile("org.quartz-scheduler:quartz:2.2.3")
+    compile("org.quartz-scheduler:quartz:2.3.1")
```
Comment (pwojnowski) [inline build.gradle]: 2.3.2 has been released recently. Maybe we could use it. WDYT?

Comment (Chevron94) [reply build.gradle]: Quartz was updated to 2.3.2

**Comment by pwojnowski** [comment]
Great job, @Chevron94 ! :+1: Thank you very much!

**Pull request merged**